internal/controllers/core/kubernetesdiscovery/container_restart.go

```diff
@@ -53,8 +53,8 @@ func (c *ContainerRestartDetector) logRestarts(dispatcher Dispatcher, mn model.M
 
 func podsByUID(status v1alpha1.KubernetesDiscoveryStatus) map[types.UID]*v1alpha1.Pod {
 	pods := make(map[types.UID]*v1alpha1.Pod)
-	for _, p := range status.Pods {
-		pods[types.UID(p.UID)] = &p
+	for i := range status.Pods {
+		pods[types.UID(status.Pods[i].UID)] = status.Pods[i].DeepCopy()
```

Comment:
Naw, linter would have been happy with what you wrote too.

This was a "free" optimization; since I wanted to use `DeepCopy()`, it made since to eliminate the intermediate copy-by-value from the `range` iteration variable IMO